src/Microsoft.Health.Fhir.Liquid.Converter/Filters/DateFilters.cs

```diff
@@ -48,28 +48,33 @@ public static string FormatAsDateTime(string input)
             }
 
             var groups = matches[0].Groups;
-            if (groups["time"].Success)
+            if (!groups["time"].Success)
             {
-                // Convert DateTime with time zone
-                string result = ConvertDateTime(input, groups);
-                if (groups["timeZone"].Success)
-                {
-                    int timeZoneHour = int.Parse(groups["timeZoneHour"].Value);
-                    int timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value);
-                    result += groups["sign"].Value + new TimeSpan(timeZoneHour, timeZoneMinute, 0).ToString(@"hh\:mm");
-                }
-                else
-                {
-                    result += "Z";
-                }
+                return ConvertDate(input, groups);
+            }
 
-                return result;
+            // Convert DateTime with time zone
+            var result = ConvertDateTime(input, groups);
+            if (groups["timeZone"].Success)
+            {
+                var timeZoneHour = int.Parse(groups["timeZoneHour"].Value);
+                var timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value);
+                result += groups["sign"].Value + new TimeSpan(timeZoneHour, timeZoneMinute, 0).ToString(@"hh\:mm");
```

Comment:
Maybe we can put this logic into ConvertDateTime function and create datetime with TimeSpan in constructor ( rather than string concat)?
Than we don't need to TryParse again to validate result.

Reply:
Thanks for the comment! As this is a PR from dotliquid to main, we can see if we can refine it in later PRs. :)